code42/py42
# Feature/more exceptions

### Description of Change ###

    - `Py42UserAlreadyAddedError`
    - `Py42UserNotInLegalHoldError`
    - `Py42LegalHoldNotFoundOrPermissionDeniedError`
    - `Py42UserDoesNotExistError`
    - `Py42InvalidRuleTypeError`

And their logic mirrored from what is in the CLI.
The CLI tests still pass as well.

### Issues Resolved ###
INTEG-1072

### Testing Procedure ###
Verify these statements:
- `py42.detectionlists.departing_employee.add()` now raises `Py42UserAlreadyAddedError` when the user is already on the list.
- `py42.detectionlists.high_risk_employee.add()` now raises `Py42UserAlreadyAddedError` when the user already on the list.
- `py42.legalhold.add_to_matter()` now raises `Py42UserAlreadyAddedError` when the user is already on the matter.
- `py42.legalhold.get_matter_by_uid()` now raises `Py42LegalHoldNotFoundOrPermissionDeniedError` when the matter is unavailable.
- `py42.users.get_by_username()` now raises `Py42UserDoesNotExistError`.
- `py42.alerts.rules.add_user()` now raises `Py42InvalidRuleTypeError` when trying to add a user to a system rule.
- `py42.alerts.rules.remove_user()` now raises `Py42InvalidRuleTypeError` when trying to a remove user from a system rule.

### PR Checklist ###
Did you remember to do the below?
